src/models/templates/vaccine.ts

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable camelcase */
+/* copied from https://github.com/Notarise-gov-sg/api-notify/blob/master/src/model/notification/index.ts */
+import { Static, String, Literal, Array, Record, Union } from "runtypes";
+import { Vaccination } from "../../types";
+import { config } from "../../config";
+
+const channel_mode = Union(Literal("SPM"), Literal("SPMORSMS"));
+const delivery = Union(Literal("IMMEDIATE"), Literal("SCHEDULE"));
+
+const templateInputV4 = Record({
+  qrissuedts: String,
+  qrexpiryts: String,
+  name: String,
+  travelpassport: String,
+  vaccname: String,
+  vaccdate: String,
+  qrcode: String, // base64 qrcode content
+});
+
+const templateV4 = Record({
+  template_id: Literal("SAFETRAVEL-QR-NTF-04"),
+  template_input: templateInputV4,
+});
+const template_layout = Array(Union(templateV4));
+
+const cta = Record({
+  action_name: String,
+  action_url: String,
+  action_type: String,
+});
+const title = String;
+const sender_name = String;
+const sender_logo_small = String; // URL of logo
+const category = Literal("MESSAGES");
+const priority = Literal("HIGH");
+const uin = String; // NRIC or FIN
+const notification_req = Record({
+  uin,
+  channel_mode,
+  delivery,
+  template_layout,
+  cta: Array(cta),
+  title,
+  sender_name,
+  sender_logo_small,
+  category,
+  priority,
+});
+
+const spmPayloadDef = Record({
+  notification_req,
+});
+
+export type SpmPayload = Static<typeof spmPayloadDef>;
+
+type SpmTemplate = Static<typeof templateV4>;
+
+export const getSpmTemplateV4 = (
+  name: string,
+  qrCode: string,
+  passportNumber: string,
+  vaccinations: Vaccination[],
+  validFrom: string
+): SpmTemplate => {
+  const { validityInDays } = config.vaccination;
+  const expiryts = new Date(validFrom);
+  expiryts.setDate(expiryts.getDate() + validityInDays);
+
+  return {
+    template_id: "SAFETRAVEL-QR-NTF-04",
+    template_input: {
+      qrissuedts: validFrom,
+      qrexpiryts: expiryts.toISOString(),
+      name,
+      travelpassport: passportNumber,
+      vaccname: vaccinations[0].vaccineName,
+      vaccdate: vaccinations[0].vaccinationDateTime,
```

Comment:
i think this one shld be the vaccine effective date

Reply:
Vaccine effective date refers to `unwrappedVaccinationCert.validFrom`?

Reply:
shld be from fhirBundle with resourceType `ImmunizationRecommendation` under `dateCriterion`

Reply:
Oh okay. In that case, I will need to require another field in `notifyVaccine()` to request for `dateCriterion`.

```javascript
// src/models/notifyVaccine/notifyVaccine.ts

interface NotifyVaccineProps {
  name: string;
  url: string;
  nric?: string;
  passportNumber: string;
  vaccinations: Vaccination[];
  validFrom: string;
  dateCriterion: string;  // To add
}
```

Since we're on this topic, what does `Vaccination.vaccinationDateTime` refers to?

Reply:
its when the vaccination is taken
e.g.
- 1st dose on 01/03/2021 (Vaccination[0].vaccinationDateTime)
- 2nd dose on 22/03/2021 (Vaccination[1].vaccinationDateTime)
- effective on 05/04/2021 (ImmunizationRecommendation dateCriterion)

Reply:
nit: instead of calling it `dateCriterion`, maybe for NotifyVaccineProps, we can name it `vaccinationEffectiveDate`/`vaccinationEffectiveFrom`, etc?

Reply:
Sure, I've edited the above comment to use `vaccinationEffectiveDate`